### PR TITLE
EZEE-3485: ezcommerce 3.3 compatibility & schema definition

### PR DESCRIPTION
**Pull request by Steveb-p**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [IBX-84](https://issues.ibexa.co/browse/IBX-84)
| **Type**                                   | improvement
| **Target Ibexa version** | `v3.3`

## Check the companion PR https://github.com/ezsystems/ezcommerce-rest/pull/10 for code changes.

This PR adds missing ecommerce REST API documentation.

### CODE REVIEW

File: input/ez-commerce.raml [hunk 1]
```diff
@@ -0,0 +1,289 @@
+displayName: eCommerce API
+# Known issue: Content-Type header is incorrectly set to "text/html" in each response.
+
+/basket:
+    get:
+        displayName: Reads list of baskets. Returns wishlist/stored baskets.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
        displayName: Read list of baskets
```

File: input/ez-commerce.raml [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        description: |
+            Returns baskets for the current user.
+            <br />Warning: Only "storedBasket" and "wishList" types are handled. Others will return validation error.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
            Returns baskets for the current user. Only "storedBasket" and "wishList" types are handled. Others will return validation error.
```

File: input/ez-commerce.raml [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+                                example: !include examples/commerce/basket/Basket.json.example
+                    400:
+                        body:
+                            application/vnd.ez.api.ValidationResponse+json:
+                                type: ValidationResponse
+                                example: !include examples/commerce/basket/current/copyfrom/_basket_id_/POST/ValidationResponse.json.example
+
+        /party/{partyType}:
+            patch:
+                displayName: Update party (buyer, delivery, invoice) information in the basket.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                displayName: Update party information in the basket
```

File: input/ez-commerce.raml [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+            responses:
+                200:
+                    body:
+                        application/vnd.ez.api.Basket+json:
+                            type: Basket
+                            example: !include examples/commerce/basket/_id_/GET/Basket.json.example
+
+        /{mode}:
+            post:
+                displayName: Update the name or the note of the stored basket.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                displayName: Update the name or the note of the stored basket
```

File: input/ez-commerce.raml [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+                    200:
+                        body:
+                            application/vnd.ez.api.Basket+json:
+                                type: Basket
+                                example: !include examples/commerce/basket/Basket.json.example
+
+        /{basketId}:
+            /line:
+                post:
+                    displayName: Add lines to basket fields for item data, provided via POST data and parsed
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                    displayName: Add lines to basket fields
```

File: input/ez-commerce.raml [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+                400:
+                    body:
+                        application/vnd.ez.api.ValidationResponse+json:
+                            type: ValidationResponse
+                            example: !include examples/commerce/basket/POST/ValidationResponse.json.example
+
+/checkout:
+    /paymentmethods:
+        get:
+            displayName: Gets list of payment methods.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
            displayName: Get list of payment methods
```

File: input/ez-commerce.raml [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+            responses:
+                200:
+                    body:
+                        application/vnd.ez.api.PaymentMethodDataResponse+json:
+                            type: PaymentMethodDataResponse
+                            example: !include examples/commerce/checkout/payment-methods/GET/PaymentMethodDataResponse.json.example
+
+    /shippingmethods:
+        get:
+            displayName: Gets list of shipping methods.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
            displayName: Get list of shipping methods
```

File: input/ez-commerce.raml [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+        patch:
+            displayName: Save the external reference number in basket
+
+    /update-remark:
+        patch:
+            displayName: Save the remark in basket
+
+/customer/addresses/shipping:
+    get:
+        displayName: Gets list of shipping addresses from customer.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
        displayName: Get list of shipping addresses
```

File: input/ez-commerce.raml [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+                                type: Basket
+                                example: !include examples/commerce/basket/Basket.json.example
+                    400:
+                        body:
+                            application/vnd.ez.api.ValidationResponse+json:
+                                type: ValidationResponse
+
+        /voucher:
+            patch:
+                displayName: Update and checks voucher in current basket
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                displayName: Update and check voucher in current basket
```

File: input/ez-commerce.raml [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+                description: Update and checks voucher in current basket
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                description: Updates and checks voucher in current basket.
```

File: input/ez-commerce.raml [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+                    body:
+                        application/vnd.ez.api.Basket+json:
+                            type: Basket
+                            example: !include examples/commerce/basket/Basket.json.example
+
+        /copyfrom/{basketId}:
+            post:
+                displayName: Copy basket to user session
+                description: |-
+                    This adds basket lines into current basket in user session, if any, using basket stored in permanent
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                    Adds basket lines into current basket in user session, if any, using basket stored in permanent
```

File: input/ez-commerce.raml [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+                                example: !include examples/commerce/basket/Basket.json.example
+                    400:
+                        body:
+                            application/vnd.ez.api.ValidationResponse+json:
+                                type: ValidationResponse
+
+        /shippingmethod:
+            patch:
+                displayName: Update shipping information in current basket
+                description: Update shipping information in current basket
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                description: Updates shipping information in the current basket.
```

File: input/ez-commerce.raml [hunk 13]
```diff
@@ -0,0 +1,289 @@
...
+                    400:
+                        body:
+                            application/vnd.ez.api.ValidationResponse+json:
+                                type: ValidationResponse
+                                example: !include examples/commerce/basket/current/copyfrom/_basket_id_/POST/ValidationResponse.json.example
+
+        /party/{partyType}:
+            patch:
+                displayName: Update party (buyer, delivery, invoice) information in the basket.
+                description: Update party (buyer, delivery, invoice) information in the basket.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                description: Updates party (buyer, delivery, invoice) information in the basket.
```

File: input/ez-commerce.raml [hunk 14]
```diff
@@ -0,0 +1,289 @@
...
+                                example: !include examples/commerce/basket/Basket.json.example
+                    400:
+                        body:
+                            application/vnd.ez.api.ValidationResponse+json:
+                                type: ValidationResponse
+
+        /paymentmethod:
+            patch:
+                displayName: Update payment information in current basket
+                description: Update payment information in current basket
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                description: Updates payment information in the current basket.
```

File: input/ez-commerce.raml [hunk 15]
```diff
@@ -0,0 +1,289 @@
...
+                                example: !include examples/commerce/basket/Basket.json.example
+                    400:
+                        body:
+                            application/vnd.ez.api.ValidationResponse+json:
+                                type: ValidationResponse
+
+        /line:
+            post:
+                displayName: Add products to current basket
+                description: Add products to current basket
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                description: Adds products to the current basket.
```

File: input/ez-commerce.raml [hunk 16]
```diff
@@ -0,0 +1,289 @@
...
+                    400:
+                        body:
+                            application/vnd.ez.api.ValidationResponse+json:
+                                type: ValidationResponse
+                                example: !include examples/commerce/basket/current/lines/POST/ValidationResponse.json.example
+
+    /{basketId}:
+        get:
+            displayName: Get Basket by ID
+            description: Get Basket by ID
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
            description: Gets basket by ID.
```

File: input/ez-commerce.raml [hunk 17]
```diff
@@ -0,0 +1,289 @@
...
+                    body:
+                        application/vnd.ez.api.Basket+json:
+                            type: Basket
+                            example: !include examples/commerce/basket/_id_/GET/Basket.json.example
+
+        /{mode}:
+            post:
+                displayName: Update the name or the note of the stored basket.
+                description: |
+                    Update the name or the note of the stored basket.
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
                    Updates the name or the note of the stored basket.
```

File: input/ez-commerce.raml [hunk 18]
```diff
@@ -0,0 +1,289 @@
...
+                        400:
+                            body:
+                                application/vnd.ez.api.ValidationResponse+json:
+                                    type: ValidationResponse
+                                    example: !include examples/commerce/basket/_id_/line/_line_id_/DELETE/ValidationResponse.json.example
+
+    /{basketName}:
+        post:
+            displayName: Create a new persisted basket
+            description: Create a new persisted basket
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
            description: Creates a new persisted basket.
```

File: input/ez-commerce.raml [hunk 19]
```diff
@@ -0,0 +1,289 @@
...
+            200:
+                body:
+                    application/json:
+                        type: ShippingAddressesResponse
+                        example: !include examples/commerce/customer/addresses/shipping/GET/ShippingAddressesResponse.json.example
+
+/customerprice:
+    post:
+        displayName: Get customer prices for the requested products
+        description: Get customer prices for the requested products
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
        description: Gets customer prices for the requested products.
```

File: input/ez-commerce.raml [hunk 20]
```diff
@@ -0,0 +1,289 @@
...
+        responses:
+            200:
+                body:
+                    application/json:
+                        type: ShippingAddressesResponse
+                        example: !include examples/commerce/customer/addresses/shipping/GET/ShippingAddressesResponse.json.example
+
+/customerprice:
+    post:
+        displayName: Get customer prices for the requested products
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
        displayName: Get customer prices for products
```

File: input/ez-commerce.raml [hunk 21]
```diff
@@ -0,0 +1,289 @@
...
+                body:
+                    application/json:
+                        type: PriceResponse
+                        example: !include examples/commerce/customerprice/POST/PriceResponse.json.example
+
+
+/common/check_sku_file:/{mode}:
+    post:
+        displayName: Add to basket the line from a CSV or Excel file
+        description: Add to basket the line from a CSV or Excel file
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
        description: Adds to basket a line from a CSV or Excel file.
```

File: input/ez-commerce.raml [hunk 22]
```diff
@@ -0,0 +1,289 @@
...
+            200:
+                body:
+                    application/json:
+                        type: PriceResponse
+                        example: !include examples/commerce/customerprice/POST/PriceResponse.json.example
+
+
+/common/check_sku_file:/{mode}:
+    post:
+        displayName: Add to basket the line from a CSV or Excel file
```
Comment (DominikaK) [inline input/ez-commerce.raml]:
```suggestion
        displayName: Add to basket from CSV or Excel
```

**Review by DominikaK — CHANGES_REQUESTED** [review]
Reminder: add generated index.md before merging

**Review by Nattfarinn — APPROVED** [review]

**Review by DominikaK — APPROVED** [review]

**Pull request merged**